runtime/lua/vim/treesitter.lua

```diff
@@ -1,44 +1,21 @@
 local api = vim.api
-local LanguageTree = require('vim.treesitter.languagetree')
-local Range = require('vim.treesitter._range')
 
 ---@type table<integer,LanguageTree>
 local parsers = setmetatable({}, { __mode = 'v' })
 
----@class vim.treesitter
----@field highlighter vim.treesitter.highlighter
----@field query vim.treesitter.query
----@field language vim.treesitter.language
```

Comment:
As extra cleanup, doesn't need to be done on this PR. But these classes are no longer needed `vim.treesitter.highlighter`, `vim.treesitter.query` and `vim.treesitter.language` since we are now using `@module` to pull the type information.